tests/base/src/test/java/org/wildfly/security/http/basic/BasicAuthenticationMechanismTest.java

```diff
@@ -52,6 +52,10 @@ public void testBasic(String username, String realm, String password, String aut
         request = new TestingHttpServerRequest(new String[] {authorization});
         mechanism.evaluateRequest(request);
         Assert.assertEquals(AbstractBaseHttpTest.Status.COMPLETE, request.getResult());
+
+        // request with incorrect password, it should be 404 and response added
+        response = request.getResponse();
```

Comment:
One thing to watch out for here is that the `testBasic` method can now be called for two different cases. One case is where the provided password is correct. The second case is now where the provided password is incorrect. To be able to control which assertions are checked for the two cases, we'll want to introduce a new parameter (e.g., `boolean wrongPassword`), e.g., something like this:

```
if (wrongPassword) {
     response = request.getResponse();
     Assert.assertEquals(HttpConstants.UNAUTHORIZED, response.getStatusCode());
} else {
    Assert.assertEquals(AbstractBaseHttpTest.Status.COMPLETE, request.getResult());
}
```

To simplify things so that you don't have to update existing calls to `testBasic` to add the new parameter, you could also introduce a new helper method that sets `wrongPassword` to false by default, i.e.,

```
private void testBasic(String username, String realm, String password, String authorization) throws Exception {
     testBasic(username, realm, password, authorization, false);
}
```
That way existing calls to `testBasic` will set `wrongPassword` to `false`.

Your new calls to `testBasic` below can then be updated to set `wrongPassword` to `true`.